### DISCUSSION TITLE
Feature request: Server side rendering

### DISCUSSION THREAD

**Discussion by Immortalin** [idom-team/idom#486] [Ideas]
**Is your feature request related to a problem? Please describe.**
A clear and concise description of what the problem is. Ex. I'm always frustrated when the page takes time to load, especially when the widgets are big.

**Describe the solution you'd like**
Server side rendering like those used by Gatsby and Next.js.

**Describe alternatives you've considered**
N/A
**Additional context**
Rehydrating the websocket connection will probably require custom code. SSR will most likely need node.js as a dependency.

**Comment by rmorshea** [comment]
@Immortalin I'm definitely open to this since it would help to address questions [like this](https://twitter.com/MarcSkovMadsen/status/1363447679179780097). However, for the moment I'm both under informed on the topic and over committed to other work.

Unless I can find some help on this one, I'll have to finish up what I've already promised to do with IDOM (as well as fix bugs) and then educated myself about Gatsby/Next.js.

Perhaps an intermediate solution to loading times would be to figure out some clever compression techniques to reduce the dependence on network latency?

**Comment by Archmonger** [comment]
> @Immortalin I'm definitely open to this since it would help to address questions [like this](https://twitter.com/MarcSkovMadsen/status/1363447679179780097). However, for the moment I'm both under informed on the topic and over committed to other work.
> 
> Unless I can find some help on this one, I'll have to finish up what I've already promised to do with IDOM (as well as fix bugs) and then educated myself about Gatsby/Next.js.
> 
> Perhaps an intermediate solution to loading times would be to figure out some clever compression techniques to reduce the dependence on network latency?

The simplest SSR method would involve rendering any new DOM components as a HTML string and pushing them into the `innerHTML` of the component selector.

**Comment by Archmonger** [comment]
@rmorshea Might be worth moving all feature requests to GH Discussions so people can vote on them.

**Comment by Archmonger** [comment]
@rmorshea Should manually remove this from the [0.33 release notes](https://github.com/idom-team/idom/releases/tag/0.33.0). Although technically the ticket was closed, having it on the changelog it makes it look like SSR capabilities were completed.

**Comment by rmorshea** [comment]
This feature will be discussed in #578
